src/index.js

```diff
@@ -96,23 +128,98 @@ class LeaderboardApp {
     return table;
   }
 
+  // eslint-disable-next-line class-methods-use-this
   createTableHeader(text) {
     const th = document.createElement('th');
     th.textContent = text;
     return th;
   }
 
-  createScoreForm() {
+  // we will add a new method submitScore which will send POST request to the API to save the score
+  async submitScore(gameId, userName, score) {
+    const url = `https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/${gameId}/scores/`;
+
+    const data = {
+      user: userName,
+      score: Number(score), // convert the score into number
+    };
+
+    try {
+      const response = await fetch(url, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(data),
+      });
+
+      if (!response.ok) {
+        throw new Error('Failed to submit score.');
+      }
+
+      // Update the list of scores after sending the score
+      await this.updateScores(gameId);
+      return response.json();
+    } catch (error) {
+      throw new Error('Failed to submit score.');
+    }
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  async updateScores(gameId) {
+    const url = `https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/${gameId}/scores/`;
+
+    try {
+      const response = await fetch(url);
+      const data = await response.json();
+
+      // Update the list of scores on the page
+      const ul = document.querySelector('.nameList');
+      ul.innerHTML = ''; // Clear the current list to update it
+
+      data.result.forEach((score, index) => {
+        const li = document.createElement('li');
+        li.textContent = `Name: ${score.user}, Score: ${score.score}`;
+        li.style.backgroundColor = index % 2 === 0 ? 'white' : 'silver'; // Alternate white and silver backgrounds
+        ul.appendChild(li);
+      });
+    } catch (error) {
+      throw new Error('Failed to update scores:', error);
+    }
+  }
+
+  // Modify createScoreForm method to call submitScore on form submission
+  createScoreForm(gameId) {
```

Comment:
You did well with your project, kindly note according to the requirement, the `refresh` button should fetch the recently updated scores but this doesn't happen in your project and instead it fetches everything in the database. Please this is an abnormal behaviour that should be corrected, kindly ensure that the `refresh` button fetches only the recently inputted values 😄 

Project Requirement :
![image](https://github.com/GhostEsso/Leaderboard/assets/20537712/4e955c05-6f69-4af5-a98a-189c7830c31c)


Your project :

https://github.com/GhostEsso/Leaderboard/assets/20537712/ee4bc5a8-4593-4559-88f2-b71c07b4fd0c